lib/logstash/inputs/syslog.rb

```diff
@@ -107,65 +109,94 @@ def run(output_queue)
   end # def run
 
   private
+  # server call the specified protocol listener and basically restarts on
+  # any listener uncatched exception
+  #
+  # @param protocol [Symbol] either :udp or :tcp
+  # @param output_queue [Queue] the pipeline input to filters queue
+  def server(protocol, output_queue)
+    self.send("#{protocol.to_s}_listener", output_queue)
```

Comment:
symbols will do an implicite conversion to string when used in string interpolation.